pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -383,6 +383,23 @@ public RebalanceResult rebalance(TableConfig tableConfig, Configuration rebalanc
     }
   }
 
+  public RebalanceResult.Status rebalanceStatus(String tableNameWithType, IdealState idealState,
+      ExternalView externalView) {
+
+    if (idealState == null) {
+      throw new IllegalStateException("ideal state is empty");
+    }
+    if (externalView == null) {
+      return RebalanceResult.Status.NO_OP;
+    }
+
+    if (isExternalViewConverged(tableNameWithType, externalView.getRecord().getMapFields(),
+        idealState.getRecord().getMapFields(), true)) {
+      return RebalanceResult.Status.DONE;
```

Comment:
I am not sure if it makes sense / helpful enough to sort of implement a stateless API to get status of rebalance task -- which was triggered separately through some other API and is being done asynchronously on potentially a different controller that this API call comes on

Relying on the fact that `ExternalView` is not the same as `IdealState` to indicate Rebalance `IN_PROGRESS` could be misleading. 

Segment assignment changes ideal state as soon as segment lands on controller and up until server successfully downloads the segment etc, `ExternalView` does not get updated. But this API will return rebalance is in `IN_PROGRESS` even though rebalance may have already finished or could have errored out etc.

So not sure if this is going to be super helpful / accurate.

Reply:
cc @antwigambrah @Jackie-Jiang thoughts ?

Reply:
HI, @siddharthteotia  Yes it will be quite misleading. Checking for IN_PROGRESS seems tricky here. 

The possible way I can think for now through testing I have done on TableRebalancerClusterStatelessTest is to check on the external view from being null to being allocated segments assignments. 

For DONE status, I will run the rebalance in dry mode and retrieve the new assignments from the result and use is checking if the extervalViewConverged . Kindly let me know you thoughts on it . cc @Jackie-Jiang